miniml/compiler/compile.scm

```diff
@@ -46,7 +47,14 @@
 (define (mkfunctor args body)
   (fold-right (lambda (arg b) (list 'MEFunctor arg b)) body args))
 
-(define ml-parser
+; note: ml-parser takes a dummy argument, because we were not able
+; to call the same lalr-parser on several input files in a reliable way.
+; (There is no documentation about this, or in general very little
+; documentation about lalr-parser.) The bug we would observe
+; is that calling (ml-parser lexer errorp) a second time after changing
+; the default input port would return an empty document. There seems
+; to be some per-parser global state that we don't know how to (re)initialize.
+(define (ml-parser _)
```

Comment:
I think you could use a no-argument function here: `(define (ml-parser) ...)`.
Also, I'm seeing a parser called several times in the documentation [here](http://wiki.call-cc.org/eggref/5/lalr), but there is probably a difference with what we are doing.

Reply:
Good idea, thanks, I made the change and rebased.